qemu/tests/balloon_boot_in_pause.py

```diff
@@ -16,14 +16,14 @@ class BallooningTestPause(BallooningTest):
     """
 
     def __init__(self, test, params, env):
-        super(BallooningTestPause, self).__init__(test, params, env)
-
+        self.params = params
```

Comment:
Not must,  why add it again ?

Reply:
Yes, it is must. It will be used in line94. Thanks

Reply:
I mean, it already be define in parent class. self.test, self.params and self.env.  No need to set it again.

Reply:
Your class should have a `test` attribute, since you don't get it from parent classes now.
When the class has it, in line 67, it can simply be `self.test.fail("...")`. In general, when you define a test class, it is better have `test`, `params` and `env` as class attributes as `MemoryBaseTest` did.

Reply:
It may be a design problem for this case as `BallooningTestPause` inherits `BallooningTest`, which inherits `MemoryBaseTest`.
In this implementation, `BallooningTestPause` only needs `run_balloon_sub_test()` from `BallooningTest`. It may need a new design of class inheritance.

By the way, ac75846c (Lucas Meneghel Rodrigues 2017-08-18 10:30:04 +0200  19)         super(BallooningTestPause, self).__init
__(test, params, env), it was added 5 month ago, not sure if this case worked after this change.

Reply:
Thanks, updated, please help review. @hereischen @luckyh